### ISSUE TITLE
consider removing dependency on meliae

### ISSUE THREAD

**Issue by anarcat** [linkchecker/linkchecker#268]
As part of the Python 2 removal in Debian, I was warned linkchecker depends on a python2-only dependency that is also dead upstream, python-meliae. I haven't investigated this further, but it seems like something we want to look into if we want to truly fix Python3 support.

See also [bug #939026](https://bugs.debian.org/939026) in Debian.

**Comment by mgedmin** [comment]
:+1:

IIRC meliae is a debugging tool (a memory profiler), so there should be no need to depend on it for anything (other than memory optimization tasks, but I don't think we're planning to do any memory optimization in the near future).

**Comment by cjmayo** [comment]
Yes, it seems clear anything Python 2 only has got to go. I just ask that we do this one after the existing PR's.
What to do though? Merge #230 as is? And then more individual PRs or tidy up #249 and merge in one go?

**Comment by anarcat** [comment]
honestly at this point, i'd do anything to get py3 in, so do as you guys feel is better.

if you need a tie-breaker, i can act as one, but otherwise let's just get something through the door and out there so people can test it. we could make a new 10.0 alpha or something and get people to get our new shiny code already! :)

**Comment by anarcat** [comment]
but that should be discussed in the other PRs.

**Comment by cjmayo** [comment]
meliae has been ported to Python 3, but there is currently a bug (reported with Python 2.7) that is stopping it working with linkchecker:

https://bugs.launchpad.net/meliae/+bug/1879100

There is a patch on the bug, which is less than a fortnight old a this time.

**Comment by mgedmin** [comment]
And, heh, bug #65 means that maybe we ought to do some memory optimization in the near future.

**Comment by anarcat** [comment]
could we consider this dependency gone now?
